### PR TITLE
feat: :sparkles: Added the search reading history API calls

### PR DESCRIPTION
**Pull request by rebelchris**
Added the search reading history API calls and test cases.

Related to:
https://github.com/dailydotdev/apps/pull/893

### CODE REVIEW

File: __tests__/__snapshots__/users.ts.snap [hunk 1]
```diff
@@ -238,6 +238,23 @@ Object {
 }
 `;
 
+exports[`query search reading history should return bookmarks search empty feed 1`] = `undefined`;
+
+exports[`query search reading history should return reading history search feed 1`] = `undefined`;
```
Comment (rebelchris) [inline __tests__/__snapshots__/users.ts.snap:243]: Working on these

File: __tests__/users.ts [hunk 2]
```diff
@@ -1351,3 +1351,88 @@ describe('mutation hideReadHistory', () => {
     expect(await repo.find()).toMatchSnapshot();
   });
 });
+
+describe('query searchBookmarksSuggestions', () => {
```
Comment (idoshamun) [inline __tests__/users.ts:1355]:
```suggestion
describe('query searchReadingHistorySuggestions', () => {
```

File: __tests__/users.ts [hunk 3]
```diff
@@ -1351,3 +1351,88 @@ describe('mutation hideReadHistory', () => {
...
+      { userId: loggedUser, timestamp: subDays(now, 4), postId: 'p4' },
+      { userId: loggedUser, timestamp: subDays(now, 5), postId: 'p5' },
+      { userId: loggedUser, timestamp: subDays(now, 6), postId: 'p6' },
+      { userId: loggedUser, timestamp: subDays(now, 7), postId: 'p1' },
+    ]);
+    const res = await client.query(QUERY('p1'));
+    expect(res.data).toMatchSnapshot();
+  });
+
+  it('should return bookmarks search empty feed', async () => {
```
Comment (idoshamun) [inline __tests__/users.ts:1424]:
```suggestion
  it('should return reading history search empty feed', async () => {
```

File: src/entity/View.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { Post } from './Post';
 
 @Entity()
 @Index(['postId', 'userId'])
+@Index(['postId', 'userId', 'hidden'])
```
Comment (idoshamun) [inline src/entity/View.ts:6]: Note: Once merged we should monitor carefully when it goes live. We need to have a look at the database

**Review by idoshamun — CHANGES_REQUESTED** [review]
Nice!!

**Review by idoshamun — APPROVED** [review]
Woohoo! Let me know once you merge

**Pull request merged**